app/lib/cluster_viz_service.rb

```diff
@@ -225,106 +242,13 @@ def self.load_cluster_group_data_array_points(study, cluster, annotation, subsam
       user_annotation = UserAnnotation.find(annotation[:id])
       subsample_annotation = user_annotation.formatted_annotation_identifier
       annotation_array = user_annotation.concatenate_user_data_arrays(annotation[:name], 'annotations', subsample_threshold, subsample_annotation)
-      x_array = user_annotation.concatenate_user_data_arrays('x', 'coordinates', subsample_threshold, subsample_annotation)
-      y_array = user_annotation.concatenate_user_data_arrays('y', 'coordinates', subsample_threshold, subsample_annotation)
-      z_array = user_annotation.concatenate_user_data_arrays('z', 'coordinates', subsample_threshold, subsample_annotation)
-      cells = user_annotation.concatenate_user_data_arrays('text', 'cells', subsample_threshold, subsample_annotation)
     else
       # for study-wide annotations, load from study_metadata values instead of cluster-specific annotations
       metadata_obj = study.cell_metadata.by_name_and_type(annotation[:name], annotation[:type])
       annotation_hash = metadata_obj.cell_annotations
-      annotation[:values] = AnnotationVizService.sanitize_values_array(metadata_obj.values, annotation[:type])
+      annotation_array = cells.map { |cell| annotation_hash[cell] }
```

Comment:
We still need to call `AnnotationVizService.sanitize_values_array` here so that we don't end up with blank values (that method interpolates `--Unspecified--` in for blank entries)

Reply:
yes, that gets taken care of just before the return

Reply:
Ah, I see it now.  Small quibble - we're reassigning the variable to itself after sanitizing it.  It doesn't really make a difference from a functional standpoint, but I think it's safer to rename the sanitized array something else.

Reply:
Good point, since this is at the end of the function, we actually don't need a variable for it at all